jbmc/unit/java_bytecode/java_bytecode_language/context_excluded.cpp

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************\
+
+Module: Unit tests for java_bytecode_language.
+
+Author: Diffblue Limited.
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <util/symbol_table.h>
+
+#include <java-testing-utils/load_java_class.h>
+#include <java-testing-utils/require_type.h>
+
+SCENARIO(
+  "Exclude a method using context-include/exclude",
+  "[core][java_bytecode][java_bytecode_language]")
+{
+  GIVEN("A class with all methods excluded except for main")
+  {
+    const symbol_tablet symbol_table =
+      load_goto_model_from_java_class(
+        "ClassWithDifferentModifiers",
+        "./java_bytecode/java_bytecode_language",
+        {},
+        {{"context-include", "ClassWithDifferentModifiers.main"}})
+        .get_symbol_table();
+
+    WHEN("Converting the methods of this class")
+    {
+      THEN(
+        "Meta-information of excluded methods is preserved. Symbol values are "
+        "nil as the lazy_goto_modelt for unit tests does not generate "
+        "stub/exclude bodies.")
+      {
+        const auto &public_instance_symbol = symbol_table.lookup_ref(
+          "java::ClassWithDifferentModifiers.testPublicInstance:()I");
```

Comment:
:pick: would be better to use `lookup` and `REQUIRE` the result to be non-null :shark:

Reply:
Done. The `REQUIRE`s had to go into the `WHEN` part, is that ok?